// @pegasusheavy/linkedin-mcp/dist/types.js
import { z } from 'zod';
// LinkedIn API types
export const LinkedInProfileSchema = z.object({
    id: z.string(),
    firstName: z.string(),
    lastName: z.string(),
    headline: z.string().optional(),
    profilePictureUrl: z.string().optional(),
    vanityName: z.string().optional(),
});
export const LinkedInPostSchema = z.object({
    id: z.string(),
    author: z.string(),
    text: z.string(),
    createdAt: z.string(),
    likeCount: z.number().optional(),
    commentCount: z.number().optional(),
    shareCount: z.number().optional(),
});
export const LinkedInConnectionSchema = z.object({
    id: z.string(),
    firstName: z.string(),
    lastName: z.string(),
    headline: z.string().optional(),
    connectedAt: z.string().optional(),
});
// Profile Edit types
export const LinkedInSkillSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
});
export const LinkedInPositionSchema = z.object({
    id: z.string().optional(),
    title: z.string(),
    company: z.string(),
    description: z.string().optional(),
    startDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }),
    endDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }).optional(),
    current: z.boolean().optional(),
});
export const LinkedInEducationSchema = z.object({
    id: z.string().optional(),
    schoolName: z.string(),
    degree: z.string().optional(),
    fieldOfStudy: z.string().optional(),
    startDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }).optional(),
    endDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }).optional(),
    grade: z.string().optional(),
    activities: z.string().optional(),
});
export const LinkedInCertificationSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    authority: z.string(),
    licenseNumber: z.string().optional(),
    startDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }).optional(),
    endDate: z.object({
        year: z.number(),
        month: z.number().optional(),
    }).optional(),
    url: z.string().optional(),
});
export const LinkedInPublicationSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    publisher: z.string().optional(),
    date: z.object({
        year: z.number(),
        month: z.number().optional(),
        day: z.number().optional(),
    }).optional(),
    description: z.string().optional(),
    url: z.string().optional(),
});
export const LinkedInLanguageSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    proficiency: z.enum(['ELEMENTARY', 'LIMITED_WORKING', 'PROFESSIONAL_WORKING', 'FULL_PROFESSIONAL', 'NATIVE_OR_BILINGUAL']).optional(),
});
